functions/src/index.ts

```diff
@@ -57,6 +70,94 @@ export const finishGame = functions.https.onCall(async (data, context) => {
     status: "done",
     endedAt: finalTime,
   });
+
+  // Update ratings of players involved based on an extension of the Elo system.
+
+  // Retrieve userIds of players in the game.
+  const playersSnap = await admin
+    .database()
+    .ref(`games/${gameId}/users`)
+    .once("value");
+  const players: string[] = [];
+  playersSnap.forEach(function (childSnap) {
```

Comment:
`playersSnap` is still a snapshot at this time. I think you might need to get `playersSnap.val()` first.  What you get from`playersSnap.val()` should be an object, so you probably can't call `forEach` on it.

Reply:
Correct, but the forEach function applied to a Datasnapshot object will return the Datasnapshot objects of the top-level children, which is what I'm using here.